packages/headless/src/features/facets/category-facet-set/category-facet-set-slice.ts

```diff
@@ -251,5 +248,6 @@ function handleCategoryFacetDeselectAll(
   }
 
   slice.request.numberOfValues = slice.initialNumberOfValues;
-  handleFacetDeselectAll<CategoryFacetRequest>(slice.request);
+  slice.request.currentValues = [];
```

Comment:
I think that's good for CategoryFacet, since the values aren't sorted by score and the structure is more complex.